docs/docs/node-interface.md

```diff
@@ -11,11 +11,18 @@ The basic node data structure is as follows:
 id: String,
 children: Array[String],
 parent: String,
+// Reserved for plugins who wish to extend other nodes.
+pluginFields: Object,
 internal: {
   contentDigest: String,
   mediaType: String,
+  // A globally unique node type choosen by the plugin owner.
   type: String,
-  pluginName: String,
+  // The plugin which created this node.
+  pluginOwner: String,
+  // Stores which plugins created which fields.
+  fieldPluginOwners: Object,
```

Comment:
It was more to namespace things as "fields" is a pretty common name nodes might want to use... agree/disagree? And yeah, I dislike how long it is plus it's heavier conceptually which I dislike too.